experimental/examples/imageviewer/shared/src/androidMain/kotlin/example/imageviewer/view/CameraView.android.kt

```diff
@@ -102,38 +109,50 @@ private fun CameraWithGrantedPermission(
+                                description = nameAndDescription.description,
+                                gps = gpsPosition
+                            ),
+                            AndroidStorableImage(imageBitmap)
+                        )
+                        capturePhotoStarted = false
+                    }
+                    LocationServices.getFusedLocationProviderClient(context)
+                        .getCurrentLocation(CurrentLocationRequest.Builder().build(), null)
+                        .apply {
+                            addOnSuccessListener {
+                                sendToStorage(GpsPosition(it.latitude, it.longitude))
+                            }
+                            addOnFailureListener {
+                                sendToStorage(GpsPosition(0.0, 0.0))
+                            }
+                        }
+                }
+
                 capturePhotoStarted = true
                 imageCapture.takePicture(executor, object : OnImageCapturedCallback() {
                     override fun onCaptureSuccess(image: ImageProxy) {
                         val byteArray: ByteArray = image.planes[0].buffer.toByteArray()
                         val imageBitmap = byteArray.toImageBitmap()
                         image.close()
-                        fun sendToStorage(gpsPosition: GpsPosition) {
-                            onCapture(
-                                createCameraPictureData(
-                                    name = nameAndDescription.name,
-                                    description = nameAndDescription.description,
-                                    gps = gpsPosition
-                                ),
-                                AndroidStorableImage(imageBitmap)
-                            )
-                            capturePhotoStarted = false
-                        }
-
-                        val lastLocation: Task<Location> =
-                            LocationServices.getFusedLocationProviderClient(context)
-                                .getCurrentLocation(
-                                    CurrentLocationRequest.Builder().build(),
-                                    null
-                                )
-                        lastLocation.addOnSuccessListener {
-                            sendToStorage(GpsPosition(it.latitude, it.longitude))
-                        }
-                        lastLocation.addOnFailureListener {
-                            sendToStorage(GpsPosition(0.0, 0.0))
-                        }
+                        addLocationInfoAndReturnResult(imageBitmap)
                     }
                 })
+                viewScope.launch {
+                    // TODO: There is known issue with Android emulator
+                    //  https://partnerissuetracker.corp.google.com/issues/161034252
+                    //  After 5 seconds delay, let's assume that bug appear and publish prepared photo
```

Comment:
TODO: There is **a** known issue with Android emulator
                    //  After 5 seconds delay, let's assume that **the** bug appear**s** and publish **a** prepared photo

Reply:
Done